core/Network/TLS/Extension.hs

```diff
@@ -262,8 +262,9 @@ decodeServerName = runGetMaybe $ do
   where
     getServerName = do
         ty    <- getWord8
-        sname <- getOpaque16
-        let name = case ty of
+        snameParsed <- getOpaque16
```

Comment:
What about inserting `B.copy` to `getOpaque16` (all functions using `getBytes`) intead of here?

Reply:
That would certainly work, but we have also many transient elements, only retained on the stack, or in the context shortly, for which we don't need copying. This is why I clear the cookie value for example.